tests/python/contrib/test_hexagon/benchmark_util.py

```diff
@@ -0,0 +1,148 @@
+
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import csv
+
+class BenchmarksTable:
+    """
+    Stores/reports the result of benchmark runs.
+
+    Each line item has a status: success, fail, or skip.
+
+    Each 'success' line item must include benchmark data,
+    in the form provided by TVM's `time_evaluator` mechanism.
+
+    Each line item may also specify values for any subset of
+    the columns provided to the table's construstor.
+    """
+    BUILTIN_COLUMN_NAMES = set([
+            "row_status",
+            "timings_min_usecs",
+            "timings_max_usecs",
+            "timings_median_usecs",
+            "timings_mean_usecs",
+            "timings_stddev_usecs",
+            ])
+
+    def __init__(self):
+        self._line_items = []
+
+    def validate_user_supplied_kwargs(self, kwarg_dict):
+        name_conflicts = set(kwarg_dict.keys()).intersection(self.BUILTIN_COLUMN_NAMES)
```

Comment:
Nit: `set(kwarg_dict)` instead of `set(kwarg_dict.keys())`.  `set` takes an iterable, and `for key in dict` is common enough in python not to need the extra specificity.  This is less of an issue now that `.keys()` returns a view instead of a copied list (changed in python3, I think?), but passing a dictionary directly into `list`, `set`, `sorted`, `tuple`, etc is a pretty common idiom.